bamboo-specs/bamboo.yml

```diff
@@ -0,0 +1,243 @@
+    - Build and Test JDK 21
+    - Build and Test JDK 24
+    - Build and Test JDK 17
+- Deploy:
+    manual: false
+    final: false
+    jobs:
+    - Deploy to Maven
+- Release:
+    manual: true
+    final: false
+    jobs:
+    - Release to Maven
+Build and Test JDK 8:
+  key: BTJ8
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker run -v m2-repo:/root/.m2/repository -v ${PWD}:/module --rm -w="/module" maven:3.9.9-amazoncorretto-8 bash -c 'mvn clean package && chmod -R 777 .'
+      description: Build and test
+  artifact-subscriptions: []
+Build and Test JDK 11:
+  key: BTJ11
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker run -v m2-repo:/root/.m2/repository -v ${PWD}:/module --rm -w="/module" maven:3.9.9-amazoncorretto-11 bash -c 'mvn clean package && chmod -R 777 .'
+      description: Build and test
+  artifact-subscriptions: []
+Build and Test JDK 21:
+  key: BTJ21
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker run -v m2-repo:/root/.m2/repository -v ${PWD}:/module --rm -w="/module" maven:3.9.9-amazoncorretto-21 bash -c 'mvn clean package && chmod -R 777 .'
+      description: Build and test
+  artifact-subscriptions: []
+Build and Test JDK 24:
+  key: BTJ24
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker run -v m2-repo:/root/.m2/repository -v ${PWD}:/module --rm -w="/module" maven:3.9.9-amazoncorretto-11 bash -c 'mvn clean package && chmod -R 777 .'
+      description: Build and test
+  artifact-subscriptions: []
+Build and Test JDK 17:
+  key: BTJ17
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker run -v m2-repo:/root/.m2/repository -v ${PWD}:/module --rm -w="/module" maven:3.9.9-amazoncorretto-17 bash -c 'mvn clean package && chmod -R 777 .'
+      description: Build and test
+  artifact-subscriptions: []
+Deploy to Maven:
+  key: DTM
+  tasks:
+  - checkout:
+      force-clean-build: 'false'
+      description: Checkout default repository
+  - script:
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash -eu
+
+        set -x
+
+        docker pull ${bamboo.build.docker.image.id}
```

Comment:
Do we need to pull?

Reply:
Yes, for the reason I had share in the previous review. Basically to ensure that the image version passed in is always the latest, otherwise cached older variants could be used.

Reply:
Docker images are immutable. So if you are using a specific version `${bamboo.build.docker.image.id}`, how can it be modified and stay as the same version.

Reply:
Can you point me to the documentation where you are reading this from?

Reply:
[This is one of the docs](https://support.atlassian.com/bamboo/kb/docker-runner-doesnt-pull-the-latest-image-automatically/)...

> ... if a new version of the image being used is available, it doesn't get downloaded before the jobs run. This can cause jobs to fail if new features have been added to the image and are required for the new workflow to succeed.

> Cause
Bamboo uses a cached image on the build agent. If Docker finds the tag locally, it won't check the remote registry for an updated version of the same tag.

Reply:
Can i look at some of the values pointed at by `{bamboo.build.docker.image.id}` for the previous runs?